crypto/evp/names.c

```diff
@@ -104,9 +121,21 @@ const EVP_MAC *EVP_get_macbyname(const char *name)
     return mp;
 }
 
+const EVP_KDF *EVP_get_kdfbyname(const char *name)
```

Comment:
Should we be introducing new functions like this? Isn't this superseded by some fetch call, e.g. EVP_KDF_fetch(). This is effectively supporting "implicit fetch". Implicit fetch is something we have to do for old apps being ported to 3.0 because that's all there is in older OpenSSL versions. But do we have to support it for new algorithm types?

Reply:
This variant of the new passes an EVP_KDF object.
EVP_KDF_CTX *EVP_KDF_CTX_new(const EVP_KDF *kdf);
Wouldnt the above function be the locical way of getting an object by name?
This is the function that is currently doing the OPENSSL_init_crypto call. Can we replace it when the fetch is implemented?

Reply:
Explicit fetch looks up algorithms by name and give you an object of the requested type back. Essentially it does exactly the same thing as this function except its associated with a specific provider implementation.

Reply:
> Can we replace it when the fetch is implemented?

Perhaps yes. Please add a TODO comment to prompt us to look at this again when we have explicit fetch.